src/py/flwr/driver/app.py

```diff
@@ -127,14 +129,45 @@ def start_driver(  # pylint: disable=too-many-arguments
         initialized_config,
     )
 
+    # Request for workload_id
+    workload_id = driver.create_workload(driver_pb2.CreateWorkloadRequest()).workload_id
+    get_nodes_res = driver.get_nodes(
+        req=driver_pb2.GetNodesRequest(workload_id=workload_id)
+    )
+
+    # Register nodes
+    registered_nodes: Dict[int, DriverClientProxy] = {}
+    compare_and_update(
+        get_nodes_res,
+        driver,
+        workload_id,
+        registered_nodes,
+        initialized_server.client_manager(),
+    )
```

Comment:
I am not sure how custom ClientManager implement its `sample()` method. It is not clearly documented in the ClientManager how `sample()` works. Would it be possible that some `ClientManager.sample()` will raise an error instantly when there are not enough clients? This can be a problem because it may be likely that  `ClientManager.sample()` is called before `Driver.get_nodes()` considering multi-threading.

Reply:
The built-in implementation (`SimpleClientManager`) uses `wait_for` to wait until a minimum number of clients are connected. With that in mind, and also given that `GetNodes` can return an empty list of clients for a long time (if no clients are connected to the server yet), it's ok if the `ClientManager` gets the initial update with a slight delay.